kclvm/query/src/override.rs

```diff
@@ -240,6 +239,58 @@ struct OverrideTransformer {
 }
 
 impl<'ctx> MutSelfMutWalker<'ctx> for OverrideTransformer {
+    fn walk_module(&mut self, module: &'ctx mut ast::Module) {
+        match self.action {
+            // Walk the module body to find the target and override it.
+            ast::OverrideAction::CreateOrUpdate => {
+                module.body.iter_mut().for_each(|stmt| {
+                    if let ast::Stmt::Assign(assign_stmt) = &mut stmt.node {
+                        if assign_stmt.targets.len() == 1 && self.field_paths.len() == 0 {
+                            let target =
+                                &Some(Box::new(ast::Node::dummy_node(ast::Expr::Identifier(
+                                    assign_stmt.targets.get(0).unwrap().node.clone(),
+                                ))));
+                            let target = get_key_path(target);
+                            if target == self.target_id {
+                                let item = assign_stmt.value.clone();
+
+                                let mut value = self.clone_override_value();
+                                // Use position information that needs to override the expression.
+                                value.set_pos(item.pos());
+                                // Override the node value.
+                                assign_stmt.value = value;
+                                self.has_override = true;
+                            }
+                        }
+                    }
+                });
+            }
+            ast::OverrideAction::Delete => {
+                // Delete the override target when the action is DELETE.
+                module.body.retain(|stmt| {
+                    if let ast::Stmt::Assign(assign_stmt) = &stmt.node {
+                        if assign_stmt.targets.len() == 1 && self.field_paths.len() == 0 {
+                            let target =
+                                &Some(Box::new(ast::Node::dummy_node(ast::Expr::Identifier(
+                                    assign_stmt.targets.get(0).unwrap().node.clone(),
+                                ))));
+                            let target = get_key_path(target);
+                            if target == self.target_id {
+                                if let ast::OverrideAction::Delete = self.action {
```

Comment:
Remove the dup if expr `if let ast::OverrideAction::Delete = self.action {`

Reply:
fixed.